pulser/waveforms.py

```diff
@@ -233,6 +269,39 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f'CompositeWaveform({self.duration} ns, {self._waveforms!r})'
 
+    def __getitem__(self, index_or_slice:
+                    Union[int, slice]) -> Union[float, np.ndarray]:
+        wf_start: int
+        if isinstance(index_or_slice, slice):
+            s: slice = self._check_slice(index_or_slice)
+            start: int = s.start
+            stop: int = s.stop
+            array: np.ndarray = np.array([])
+            wf_start = 0
+            for wf in self._waveforms:
+                if wf_start <= start and start <= wf_start+wf.duration:
+                    if wf_start <= stop and stop <= wf_start+wf.duration:
+                        return cast(np.ndarray,
+                                    wf.samples[start-wf_start:stop-wf_start])
+                    else:
+                        array = wf.samples[start-wf_start:]
+                if start < wf_start and stop > wf_start+wf.duration:
+                    array = np.concatenate((array, wf.samples))
+                if wf_start <= stop and stop <= wf_start+wf.duration:
+                    array = np.concatenate((array, wf.samples[:stop-wf_start]))
+                wf_start += wf.duration
+            return array
+        else:
+            index: int = self._check_index(index_or_slice)
+            wf_start = 0
+            value: float
+            for wf in self._waveforms:
+                if wf_start <= index and index < wf_start+wf.duration:
```

Comment:
Same suggestion here:
```suggestion
                if wf_start <= index < wf_start+wf.duration:
```

Reply:
> Same suggestion here:

👍